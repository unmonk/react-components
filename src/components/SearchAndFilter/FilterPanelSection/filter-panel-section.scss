@import "~vanilla-framework/scss/base_placeholders";

.filter-panel-section {
  border-bottom: 1px solid $color-mid-x-light;
  padding-bottom: 1rem;

  &:last-child {
    border: 0;
  }

  &__heading {
    color: $color-mid-dark;
    font-size: 1rem;
    margin-bottom: 0;
  }
}
