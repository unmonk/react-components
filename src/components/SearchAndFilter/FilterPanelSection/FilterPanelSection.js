import React from "react";
import PropTypes from "prop-types";
import Chip from "../../Chip";

import "./filter-panel-section.scss";

const FilterPanelSection = ({ data }) => (
  <div className="filter-panel-section">
    {data?.heading && (
      <h3 className="filter-panel-section__heading">{data.heading}</h3>
    )}
    {data?.chips?.map((chip) => (
      <Chip
        lead={chip.lead}
        value={chip.value}
        key={`${chip.lead}+${chip.value}`}
      />
    ))}
  </div>
);

FilterPanelSection.propTypes = {
  data: PropTypes.shape({
    heading: PropTypes.string,
    chips: PropTypes.arrayOf(
      PropTypes.shape({
        lead: PropTypes.string,
        value: PropTypes.string,
      })
    ),
  }),
};

export default FilterPanelSection;
