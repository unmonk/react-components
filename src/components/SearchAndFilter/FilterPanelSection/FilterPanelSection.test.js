import { shallow, mount } from "enzyme";
import React from "react";

import FilterPanelSection from "./FilterPanelSection";

const sampleData = {
  heading: "Regions",
  chips: [{ value: "us-east1" }, { value: "us-east2" }, { value: "us-east3" }],
};

describe("Filter panel section", () => {
  it("renders", () => {
    const wrapper = shallow(<FilterPanelSection data={sampleData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("shows correct data passed as prop", () => {
    const wrapper = mount(<FilterPanelSection data={sampleData} />);
    expect(wrapper.find(".filter-panel-section").length).toEqual(1);
    const section = wrapper.find(".filter-panel-section");
    const chips = wrapper.find(".p-chip");
    expect(chips.length).toEqual(3);
    expect(section.find(".filter-panel-section__heading").text()).toEqual(
      "Regions"
    );
    const firstChip = section.find(".p-chip").at(0);
    expect(firstChip.text()).toEqual("us-east1");
  });
});
