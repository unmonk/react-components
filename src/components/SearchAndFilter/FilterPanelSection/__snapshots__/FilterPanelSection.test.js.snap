// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter panel section renders 1`] = `
<div
  className="filter-panel-section"
>
  <h3
    className="filter-panel-section__heading"
  >
    Regions
  </h3>
  <Chip
    key="undefined+us-east1"
    value="us-east1"
  />
  <Chip
    key="undefined+us-east2"
    value="us-east2"
  />
  <Chip
    key="undefined+us-east3"
    value="us-east3"
  />
</div>
`;
